src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -730,5 +783,31 @@ public ExtendedFilterResult(VSCompletionItem item, FilterResult filterResult)
                 FilterResult = filterResult;
             }
         }
+
+        private class NullablePatternMatchComparer : IComparer<ExtendedFilterResult>
+        {
+            public int Compare(ExtendedFilterResult x, ExtendedFilterResult y)
+            {
+                var matchX = x.FilterResult.PatternMatch;
+                var matchY = y.FilterResult.PatternMatch;
+
+                if (matchX.HasValue)
+                {
+                    if (matchY.HasValue)
```

Comment:
when is HasValue false?

Reply:
See this comment
https://github.com/dotnet/roslyn/pull/38499#discussion_r322415317